### ISSUE TITLE
[$4000] [iOS] - Page freezes when creating New Group and selecting 8 members

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#9809]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/docs/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Click on the green + button
3. Click on Create Group
4. Start the group creation flow and add 8 participants.
5. Scroll the page

## Expected Result:
The page should move and scroll when adding 8 members 

## Actual Result:
Page freezes when selecting 8 group members and unable to scroll the page

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS ✅ 
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.82.5

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/178056852-7b0e4aee-ee0a-4fdd-a3d1-e1b23c813198.mp4

https://user-images.githubusercontent.com/93399543/178056853-ffb30259-015a-45cf-8661-186c0b52a806.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by tylerkaraszewski** [comment]
Reproduced on iOS on my iPhone 13, does not scroll correctly.

**Comment by jliexpensify** [comment]
I suspect this is an iOS issue:

- Tested on Web, seems to be fine?

![image](https://user-images.githubusercontent.com/30815269/178172580-9422eaec-e379-49ca-8322-41c19d0cf6d7.png)

- Tested on Android, works fine?

Both were 1.1.82.5

**Comment by jliexpensify** [comment]
Posted:

Internal - https://www.upwork.com/ab/applicants/1546309325160779776/job-details
External - https://www.upwork.com/jobs/~0166ca314670548e41

**Comment by rushatgabhane** [comment]
`main` works fine on iOS and mWeb Safari.

 v1.1.79-18, iPhone 13

https://user-images.githubusercontent.com/29673073/178189987-51cb8c46-01f7-4434-ae07-a08f020c42d6.mov

**Comment by jliexpensify** [comment]
@rushatgabhane it looks like we all tested on 1.1.82.5 - can you try with that version? I believe that's newer right?

**Comment by rushatgabhane** [comment]
@jliexpensify yeah my bad, here's v 1.1.82-5

works well on safari mWeb, and iOS.

https://user-images.githubusercontent.com/29673073/178191668-a3e0235a-34eb-44c7-bb9c-b89f42034095.mov

**Comment by jliexpensify** [comment]
@rushatgabhane - awesome, thanks for testing! Might just be general lagginess happening then?

Can we close this issue, since it'll be fixed soon?

**Comment by rushatgabhane** [comment]
> Can we close this issue

@jliexpensify not just yet, we should close when QA can't repro this bug in thier weekly tests.

Because I could be wrong 😄

**Comment by jliexpensify** [comment]
Not overdue, waiting on weekly tests!

**Comment by rushatgabhane** [comment]
@kbecciv is this issue still reproducible on staging?

**Comment by jliexpensify** [comment]
I've had to take some unexpected time off, so assigning to Kadie (my leave buddy) to manage!

**Comment by jliexpensify** [comment]
TL;DR - waiting for @kbecciv to confirm if this is an issue on staging.

If not, we can close this GH!

**Comment by kbecciv** [comment]
@jliexpensify Sure, will update you shortly.

**Comment by Julesssss** [comment]
> Maybe this is a physical device only thing. @Julesssss if you have an iPhone, can you please verify?

I tested on `v1.187-9` and I'm also unable to reproduce. iPhone 7 running 13.3

@kbecciv could you share the OS version and device that your team can reproduce it on? Thanks.

**Comment by kbecciv** [comment]
Checking with team, will update you shortly

**Comment by kbecciv** [comment]
@Julesssss Please attached the version and model of the device.
![Image from iOS (43)](https://user-images.githubusercontent.com/93399543/182918399-ef2264b8-810a-412e-8f80-1ed8496422aa.png)

**Comment by Julesssss** [comment]
Sharing internally to see if anyone can reproduce with a similar device

**Comment by kadiealexander** [comment]
@Julesssss did you have any luck getting someone to repro this?

**Comment by Julesssss** [comment]
No, and I'm not sure we can do anything further until we have specific steps to narrow down the device 😕 I wonder if it has something to do with device language, as I can't think what else differs between the test devices

**Comment by kadiealexander** [comment]
@tylerkaraszewski  I saw you were able to reproduce this in the past on an iPhone 13. Could you please let us know the OS version, and whether it's still reproducible?

**Comment by Julesssss** [comment]
> Page freezes when selecting 8 group members and unable to scroll the page

@trjExpensify am I crazy, or wasn't the chat supposed to be limited to 8 participants TOTAL? 

I'm able to create a chat with 9 participants (myself included), and the UI does sort of lock in place. I wonder if that's what the initial issue was trying to describe, instead of the app actually locking up?

**Comment by Julesssss** [comment]
Okay, so this is not a bug on most devices IMO. With 8 participants selected you view and remove all of them. Perhaps this is an issue on smaller devices though, but this needs to be confirmed. 

Does anyone have a small device that restricts participant deselection?

**Comment by kadiealexander** [comment]
Same experience as Rajat for me on the ol' Asus phone. It removes the list of people you can select, but doesn't freeze - you can still deselect and take actions.

**Comment by Julesssss** [comment]
> Same experience as Rajat for me on the ol' Asus phone. It removes the list of people you can select, but doesn't freeze - you can still deselect and take actions.

Cool, so that sounds okay. 

Looking at the original reports I do think we should allow scroll when the participants are locked. But this is a pretty minor bug.

**Comment by kadiealexander** [comment]
Just adding a note that I'm OOO until September 21st. As this is on hold I won't reassign it, but if you need a contributor manager to do anything before that point please reapply the `external` label.

**Comment by kadiealexander** [comment]
Job doubled! New Upwork post:

https://www.upwork.com/ab/applicants/1584714876102766592/job-details
https://www.upwork.com/jobs/~015735edd1fac507c5

**Comment by puneetlath** [comment]
@kadiealexander I added the help wanted label since I believe we're looking for external proposals, yeah? Also, it looks like it might be time to double again.

**Comment by kadiealexander** [comment]
Sounds good! It's been a hot minute, so here's the new Upworks post doubled to $1000:

https://www.upwork.com/ab/applicants/1587903133329772544/job-details
https://www.upwork.com/jobs/~019d964b285383ca3e

**Comment by kadiealexander** [comment]
Price doubled!

**Comment by Julesssss** [comment]
The latest [update](https://github.com/Expensify/App/issues/9809#issuecomment-1235267910) for this issue is still valid

**Comment by kevinksullivan** [comment]
@kadiealexander looks like this could be doubled again?

**Comment by JmillsExpensify** [comment]
@Julesssss To help clear out the extensive backlog of /App bugs, we're putting the spotlight on all bugs older than 4 weeks old. To help unblock the roadmap and get our bug pipeline back in equilibrium, can you:

 - Decide whether any proposals currently meet our guidelines and can be approved as-is
 - For any that can't, please take this issue internal and treat it as one of your highest priorities
 - If you have any questions, don't hesitate to start a discussion in #bug-zero

Thanks everyone!

**Comment by Julesssss** [comment]
Hi @kadiealexander could you double the price again please

**Comment by kadiealexander** [comment]
Sorry, was out sick at the end of last week. Job doubled!

**Comment by railway17** [comment]
Is this issue still reproducible?
I couldn't try to test it on old versions of iOS but looks like not reproducible in newer versions.

**Comment by hellohublot** [comment]
## Proposal Append For https://github.com/Expensify/App/issues/9809#issuecomment-1321958007




This is another solution, the screenshot is same as the previous solution https://github.com/Expensify/App/issues/9809#issuecomment-1321958007, but this PR https://github.com/Expensify/App/pull/12366 removed the `cardStyle` `position` today, I also found some problems on `mobile_safari`  in that PR, such as the head of SearchScreen `cannot be fixed at the top` when scroll, such as the first time you enter the ProfileScreen page to edit the LastName, the page will move to the left, maybe should wait before he/she solve that

if we still need `cardStyle` `position` in the future, you can take a look at this solution, Thanks


https://github.com/Expensify/App/blob/6faef3fe06a35839f13e74734acdce111b13c948/src/styles/cardStyles/index.js#L9-L16

```js
export default function getCardStyles(isSmallScreenWidth, screenWidth) {
    return {
    	position: Browser.getBrowser() == CONST.BROWSER.SAFARI && Browser.isMobile() ? 'absolute' : 'fixed',
        ...
    };
}
```

```jsx
<ScrollDocument>
    <html>
        <body>
              <ReportScreen />
              <NewGroupScreen style={{ position:  isMobileSafari ? 'absolute' : 'fixed' }}>
                    <SectionList style={{ overflow: 'scroll' }} />
              </NewGroupScreen>
        <body>
    <html>
</ScrollDocument>
```

**Comment by dylanexpensify** [comment]
Amazing, just as a friendly reminder, we expect all reviews to be completed and done within 48 hours!

**Comment by marcaaron** [comment]
I'm kinda lost on this issue. Is it only affecting Safari on older iOS versions? I was not able to repro this at all on iOS 15 version of Safari.

**Comment by hellohublot** [comment]
#Q&A
## Are you on expensify's opensource slack channel yet?
I have sent the application email to `contributors@expensify.com`, my username is `hublot@aliyun.com`

## How to reproduce
You can slide the list of this page `to the top`, `then continue to drag down`, and then `repeatedly drag up and down`,
Or slide the list of this page `to the bottom`, `continue to drag up`, and then `repeatedly drag up and down`

## Why it happened
`SectionList.overflow.scroll` does not support `bounce`, so this event cannot be accepted, and this event will be sent to `Document`, because `Document always supports bounce`

## Can it be reproduced on iOS 15+
`Document bounce` has been changed to `support pull-down refresh in this version`, so this problem `cannot be reproduced`. I can’t borrow a mobile phone with iOS 15+ system. Maybe You can try to slide the list to the bottom and then drag it up 

## Can we intercept this event and hand it over to SectionList?
I have found many ways to solve this `gesture penetration`, but none of them work
Include the following.
- touchmove.stopPropagation
- document.body.position.fixed
- overscroll-behavior: contain
- webkit-overflow-scrolling: touch
- replace SectionList to View

Finally, I guess if `SectionList.overflow.scroll` can support bounce, it must be `inconsistent with w3c`, because other web cores do `not support bounce`
So I ended up `giving up on intercepting the event`

## My solution
Although we can't intercept the Document.scroll.bounce event, we can give the user `some visual feedback` instead of `the page gesture being frozen`

## First solution addEventListener('scroll')
We can listen the `offset change of Document.scroll.bounce`, and then let the entire page of NewGroupChat `scroll together`

## Second solution position: 'absolute'
This link can explain what is the difference between 'absolute' and fixed, https://www.w3.org/wiki/CSS_absolute_and_fixed_positioning?source=post_page
Simply put, absolute will not break away from the document flow, and the `pop-up window will slide along with the document flow`, but fixed `cannot slide along with the document flow`

So if we don't want the first solution, we can change NewGroupPage to `absolute`, it will `automatically scroll along with the document flow`, so we `don't have to listen to offset`

## Can we cancel the position condition and set it all to absolute?
I don't think so, because only MobileSafari supports bounce, it is special so we also treat it specially,

## If we really need to cancel the position condition, set them all to absolute. 
The code I want to debug conflicts with this PR https://github.com/Expensify/App/pull/12366. This PR caused some bugs in https://staging.new.expensify.com/, you can use MobileSafari to go Reproduce those bugs, such as ProfileScreen and SearchScreen and more, so I want to wait until they are fixed, and I also hope they can continue to use `cardStyle.position`, then I can continue to debug and answer this question

Thanks !

**Comment by trjExpensify** [comment]
>I have sent the application email to [contributors@expensify.com](mailto:contributors@expensify.com), my username is [hublot@aliyun.com](mailto:hublot@aliyun.com)

Hey @hellohublot, that email hasn't been received on our side. Can you resend it and include a link to your Upwork profile please. After which, we can get you added to Slack. Thanks!
